api/api.go

```diff
@@ -2,18 +2,55 @@ package api
 
 import (
 	"context"
+	"encoding/hex"
 	"fmt"
 	"math/big"
 
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
 	"github.com/ethereum/go-ethereum/common/math"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/eth/filters"
 	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/onflow/flow-evm-gateway/storage"
 )
 
-type BlockChainAPI struct{}
+const EthNamespace = "eth"
+const defaultGasPrice = 8049999872
+
+// TODO: Fetch these from flow-go/fvm/evm/emulator/config.go
+var (
+	FlowEVMTestnetChainID = big.NewInt(666)
+	FlowEVMMainnetChainID = big.NewInt(777)
+)
+
+// this is added to resolve the issue with chainhash ambiguous import,
+// the code is not used, but it's needed to force go.mod specify and retain chainhash version
+// workaround for issue: https://github.com/golang/go/issues/27899
```

Comment:
add a TODO that once go-ethereum is updated this can go away.

Reply:
Good catch :+1: 
I removed it altogether in https://github.com/onflow/flow-evm-gateway/pull/11/commits/eb14c1ed0df8033278a8f4983e11b9c624feb79d, since I am using the latest `go-ethereum` version, and that's going to be used in `flow-go` as well.